src/conversions/chrono.rs

```diff
@@ -41,141 +39,217 @@
     PyTzInfo, PyTzInfoAccess,
 };
+#[cfg(Py_LIMITED_API)]
+use crate::{intern, Py, PyDowncastError};
 use crate::{FromPyObject, IntoPy, PyAny, PyErr, PyObject, PyResult, Python, ToPyObject};
 use chrono::offset::{FixedOffset, Utc};
 use chrono::{
     DateTime, Datelike, Duration, NaiveDate, NaiveDateTime, NaiveTime, Offset, TimeZone, Timelike,
 };
-use std::convert::TryInto;
 
 impl ToPyObject for Duration {
     fn to_object(&self, py: Python<'_>) -> PyObject {
         // Total number of days
         let days = self.num_days();
         // Remainder of seconds
         let secs_dur = *self - Duration::days(days);
-        // .try_into() converts i64 to i32, but this should never overflow
-        // since it's at most the number of seconds per day
-        let secs = secs_dur.num_seconds().try_into().unwrap();
+        let secs = secs_dur.num_seconds();
         // Fractional part of the microseconds
         let micros = (secs_dur - Duration::seconds(secs_dur.num_seconds()))
             .num_microseconds()
             // This should never panic since we are just getting the fractional
             // part of the total microseconds, which should never overflow.
-            .unwrap()
-            // Same for the conversion from i64 to i32
-            .try_into()
             .unwrap();
 
-        // We do not need to check i64 to i32 cast from rust because
-        // python will panic with OverflowError.
-        // We pass true as the `normalize` parameter since we'd need to do several checks here to
-        // avoid that, and it shouldn't have a big performance impact.
-        let delta = PyDelta::new(py, days.try_into().unwrap_or(i32::MAX), secs, micros, true)
-            .expect("failed to construct delta");
-        delta.into()
+        #[cfg(not(Py_LIMITED_API))]
+        {
+            // We do not need to check the days i64 to i32 cast from rust because
+            // python will panic with OverflowError.
+            // We pass true as the `normalize` parameter since we'd need to do several checks here to
+            // avoid that, and it shouldn't have a big performance impact.
+            // The seconds and microseconds cast should never overflow since it's at most the number of seconds per day
+            PyDelta::new(
+                py,
+                days.try_into().unwrap_or(i32::MAX),
+                secs.try_into().unwrap(),
+                micros.try_into().unwrap(),
+                true,
+            )
+            .expect("failed to construct delta")
+            .into()
+        }
+        #[cfg(Py_LIMITED_API)]
+        {
+            timedelta_class(py)
+                .call1((days, secs, micros))
+                .expect("failed to construct datetime.timedelta")
+                .into()
+        }
     }
 }
 
 impl IntoPy<PyObject> for Duration {
     fn into_py(self, py: Python<'_>) -> PyObject {
-        ToPyObject::to_object(&self, py)
+        self.to_object(py)
```

Comment:
Why this change?

Reply:
To use a consistent style in the file.  Similary I allowed myself to put all the conversion code in `ToPyObject`  implementation and make `IntoPy<PyObject>` wrappers on top of it. If you want I can revert these changes.

Reply:
No need. I am just making sure I did not miss some semantic change here.